kitsune/search/v2/es7_utils.py

```diff
@@ -91,8 +91,11 @@ def index_object(doc_type_name, obj_id):
     doc_type = next(cls for cls in get_doc_types() if cls.__name__ == doc_type_name)
     model = doc_type.get_model()
 
-    obj = model.objects.get(pk=obj_id)
-    doc_type.prepare(obj).to_action("index")
+    try:
```

Comment:
I am curious as to why this is needed.

Reply:
Specifically here: users can exist in DB without a profile, however in elastic we index on profiles (it made more sense to me to do that, because the majority of the indexable information is in a user's profile, and I don't think we want to be indexing users without a profile). We have signals on User post_save, but there's no guarantee that a profile will exist at that point, so we may trigger a profile index for a profile which doesn't exist.

More generally: it's entirely possible that celery jobs will be executed after a row has been deleted from the DB which the task references, and I don't think that should be treated as a job failure.

Reply:
let's add a couple of lines with comments about what you added here